pkg/collector/corechecks/cluster/topologycollectors/configmap_collector.go

```diff
@@ -62,9 +68,62 @@ func (cmc *ConfigMapCollector) configMapToStackStateComponent(configMap v1.Confi
 
 	component.Data.PutNonEmpty("generateName", configMap.GenerateName)
 	component.Data.PutNonEmpty("kind", configMap.Kind)
-	component.Data.PutNonEmpty("data", configMap.Data)
+	component.Data.PutNonEmpty("data", cutData(configMap.Data, int(cmc.maxDataSize)))
 
 	log.Tracef("Created StackState ConfigMap component %s: %v", configMapExternalID, component.JSONString())
 
 	return component
 }
+
+func cutReplacement(dropped string) string {
+	hashing := sha256.New()
+	_, err := hashing.Write([]byte(dropped))
+	var hash string
+	if err != nil {
+		// doubt what error could happen, but just to satisfy linter, and in case...
+		hash = fmt.Sprintf("hash error: %v", err)
+	} else {
+		hash = hex.EncodeToString(hashing.Sum([]byte{}))[0:16]
+	}
+	return fmt.Sprintf("[dropped %d chars, hashsum: %s]", len(dropped), hash)
+}
+
+// cutData tries to reduce `data` size
+// it replaces values within the map completely or partially with `[dropped N, hash...]` string
+// it's intended to strip some big files out of configmap,
+// and on scale of hundred of KiB it's precise enough: it doesn't take into account replacement string
+// Algorithm:
+// 1. remove values that are on their own are bigger than limit
+// 2. proportionally cut the rest of the values
+func cutData(data map[string]string, maxSize int) map[string]string {
```

Comment:
I wonder if this complexity is really needed. Let's discuss with Mark.

Reply:
old implementation indeed has disadvantages (it was unstable - if one size of data in a single key is changed, then all keys were cut differently) - therefore I updated to simpler algorithm we discussed